### PR TITLE
Drush script bugs

### LINKED ISSUE #117: Directory not writeable
## Bug:
- If the themes/custom/ directory is not writeable the drush command fails abnormally

## Senario:
- Fresh D8 install on Pantheon
- Fresh kalabox copy of the Pantheon environment

## Error output:
- `kbox drush emulsify "Example Site" --machine_name="example"`
- `drush: The directory "/code/themes/custom" was not found.`
- `Drush command terminated abnormally due to an unrecoverable error.   [error]`

## Temporary Workaround:
- Create a themes/custom/ directory
- Temporarily change permissions on this directory `chmod 777`
- Change the permissions back when install is complete

### LINKED ISSUE #118: Pattern Lab Directory
## Bug:
- If a user clones this git repo down and doesn't install node modules / pattern lab they get the following error:

## Error Output:
- kbox drush emulsify "Example Site" --machine_name="example"
Source directory /code/themes/emulsify/pattern-lab is not readable or[error]
does not exist.
Successfully created the Emulsify theme "Example Site" created in: /code/themes/custom/example, you can now run 'yarn' or 'yarn install' or 'npm install' to install the node modules.

## Needs discussion:
- Should we not copy this directory during a fresh install?  It appears that when a user runs yarn within the new theme pattern lab will install properly regardless of being copied or not before.

### LINKED ISSUE #122: Need to create <theme>.libraries.yml file
@ccjjmartin I think we missed the libraries file in the theme creation script.

### PR DESCRIPTION
**Pull request by ccjjmartin**
## Purpose:
- Drush script bugs
- Fixes #117 
- Fixes #118 
- Fixes #122 

## #117 Testing:
- [x] Modify permissions on your themes directory to not be writeable
- [x] Run the normal drush script `drush emulsify "Test Theme Name"`
- [x] Verify you get output directing you on how to fix the error
- [x] Fix the permissions so you don't run into the error later

## #118 Testing:
- [x] Clone the repo down and do `NOT` run `npm install`
- [x] Run the normal drush script `drush emulsify "Test Theme Name"`
- [x] Verify you don't see an error about the pattern-lab directory not existing

## #122 Testing:
- [x] Run the slim drush script `drush emulsify "Test Theme Name" --slim`
- [x] Verify you have breakpoints.yml and libraries.yml and the desired components directories

### CODE REVIEW

**Comment by ModulesUnraveled** [comment]
@ccjjmartin I ran `drush emulsify "Test Theme Name"` as instructed, but the actual theme was just named "Test" with the machine name "test". There's no reference to "Theme Name" anywhere.

**Comment by ModulesUnraveled** [comment]
Also, when running the `--slim` variation, we need to keep some of the files/folders in the `components` directory, but not all of them. Mostly we just don't want the `_patterns` directory, but here's the list of the one's I'd probably keep
```
    'components/_data',
    'components/_macros',
    'components/_meta',
    'components/_twig-components',
    'components/css',
    'components/images',
```

**Comment by ModulesUnraveled** [comment]
Oh... actually we do need the colors directory, and everything in it. It'd also be good to scaffold the patterns directory, so add these:
```
_patterns/00-base/global/01-colors/*      // everything in here
_patterns/01-atoms
_patterns/02-molecules
_patterns/03-organisms
_patterns/04-templates
_patterns/05-pages
```

**Comment by ccjjmartin** [comment]
Need to add scripts directory

**Comment by ModulesUnraveled** [comment]
@ccjjmartin looks good. The machine name still doesn't seem to do what I expect, but the files/folders that need to be there are all there.

Expected:
`drush emulsify "Test Theme Name" --slim` produces machine name "test_theme_name"

What actually happens:
`drush emulsify "Test Theme Name --slim` produces machine name "test"

File: emulsify.drush.inc [hunk 1]
```diff
@@ -80,33 +75,81 @@ function drush_emulsify($human_readable_name = NULL) {
     $theme_path = drush_trim_path($path);
   }
 
-  drush_emulsify_create($human_readable_name, $machine_name, $description, $theme_path);
+  // Create your new theme.
+  $status = drush_emulsify_create($human_readable_name, $machine_name, $description, $theme_path);
+
+  // Notify the user of failure.
+  if ($status === FALSE) {
+    drush_set_error("EMULSIFY", "Your theme was not successfully created.");
```
Comment (JeffTomlinson) [inline emulsify.drush.inc]: Can we use single quotes here?

**Pull request merged**